rita_common/src/payment_validator/mod.rs

```diff
@@ -357,23 +426,201 @@ pub async fn validate_transaction(ts: ToValidate) {
     }
 }
 
+async fn handle_althea_tx_checking(ts: ToValidate) {
+    let althea_contact = Contact::new(
+        COSMOS_NODE_GRPC,
+        ALTHEA_CONTACT_TIMEOUT,
+        ALTHEA_CHAIN_PREFIX,
+    )
+    .unwrap();
+    // already verfied that it exists
+    let txhash = ts.payment.tx_hash.clone().unwrap();
+
+    let althea_chain_status = althea_contact.get_chain_status().await;
+    let althea_transaction = althea_contact.get_tx_by_hash(txhash.clone()).await;
+
+    match (althea_transaction, althea_chain_status) {
+        (Ok(transaction), Ok(chain_status)) => {
+            if let Some(tx_resp) = transaction.tx_response {
+                if !ts.checked {
+                    checked(ts.clone());
+                }
+
+                let tx = match tx_resp.tx {
+                    Some(a) => a.value,
+                    None => {
+                        error!("Althea chain tx {:?} has no tx field?", tx_resp);
+                        return;
+                    }
+                };
+
+                if let ChainStatus::Moving { block_height } = chain_status {
+                    // Decode TxRaw
+                    let raw_tx_any = prost_types::Any {
+                        type_url: "/cosmos.tx.v1beta1.Tx".to_string(),
+                        value: tx,
+                    };
+                    let tx_raw: TxRaw = match decode_any(raw_tx_any) {
+                        Ok(a) => a,
+                        Err(e) => {
+                            error!("Unable to decode raw_tx with {}", e);
+                            return;
```

Comment:
nicely done here, we need a lot of utility functions to make this easier though, those belong in deep space.